wgtfindreplacetext.cpp

```diff
@@ -58,21 +62,100 @@ FindReplaceText::FindReplaceText(QWidget *parent)
     lineBlockLayout->addWidget(caseSensitiveBox, 4, 1);
     lineBlockLayout->addWidget(upBtn, 5, 0);
     lineBlockLayout->addWidget(wholeWordsBox, 5, 1);
-    lineBlockLayout->addWidget(regularExprBox, 6, 1);
+
+    textReplaceLbl->setVisible(false);
+    textReplaceEdit->setVisible(false);
+    replaceBtn->setVisible(false);
+    replaceAllBtn->setVisible(false);
 
     QHBoxLayout *mainLayout = new QHBoxLayout();
     mainLayout->addLayout(lineBlockLayout);
 
     this->setWindowFlags(this->windowFlags() & ~Qt::WindowContextHelpButtonHint);
     setWindowTitle(tr("Find/Replace"));
-    setFixedSize(380, 200);
+    setFixedSize(380, 180);
     setLayout(mainLayout);
 
     connect(closeBtn, &QPushButton::clicked, this, &FindReplaceText::close);
+    connect(findBtn, &QPushButton::clicked, this, &FindReplaceText::findSlot);
+    connect(replaceBtn, &QPushButton::clicked, this, &FindReplaceText::replaceSlot);
+    connect(replaceAllBtn, &QPushButton::clicked, this, &FindReplaceText::replaceAllSlot);
+    connect(textFindEdit, &QLineEdit::textChanged, this, &FindReplaceText::isEmptyText);
+}
+
+void FindReplaceText::isReplace(bool mode)
+{
+    if (mode) {
+        textReplaceLbl->setVisible(true);
+        textReplaceEdit->setVisible(true);
+        replaceBtn->setVisible(true);
+        replaceAllBtn->setVisible(true);
+    }
+}
+
+//find text slot
+void FindReplaceText::findSlot()
+{
+    flags = setOptions();
+    newWord = textReplaceEdit->text();
+    emit findSignal(textFindEdit->text(), flags);
 }
 
+//replace button slot
+void FindReplaceText::replaceSlot()
+{
+    flags = setOptions();
+    newWord = textReplaceEdit->text();
+    emit replaceSignal(textFindEdit->text(), flags, newWord);
+}
+
+//replaceAll button slot
+void FindReplaceText::replaceAllSlot()
+{
+    flags = setOptions();
+    newWord = textReplaceEdit->text();
+    emit replaceAllSignal(textFindEdit->text(), flags, newWord);
+}
+
+//block buttons if text is empty
+void FindReplaceText::isEmptyText(QString text)
+{
+    QList<QPushButton*> list = this->findChildren<QPushButton*>();
```

Comment:
for (QPushButton *btn : this->findChildren<QPushButton*>()) {
    btn->setEnabled(!text.isEmpty());
}